src/payment/strategies/mollie/mollie-payment-strategy.ts

```diff
@@ -105,19 +120,40 @@ export default class MolliePaymentStrategy implements PaymentStrategy {
     }
 
     deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._mollieClient = undefined;
+
+        this.removeMollieComponents();
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    private removeMollieComponents(): void {
+        this._cardHolderElement?.unmount();
         this._cardNumberElement?.unmount();
-        this._expiryDateElement?.unmount();
         this._verificationCodeElement?.unmount();
-        this._cardHolderElement?.unmount();
+        this._expiryDateElement?.unmount();
 
-        return Promise.resolve(this._store.getState());
+        const mollieComponents = document.querySelectorAll('.mollie-component');
+
+        for (const i in mollieComponents) {
+            if (mollieComponents.hasOwnProperty(i)) {
+                mollieComponents[i].remove();
+            }
+        }
+
+        const controllers = document.querySelectorAll('.mollie-components-controller');
+
+        for (const i in controllers) {
```

Comment:
you can use `Object.entries()` instead of `for ...in`, that whay you dont need to check hasOwnProperty.

Reply:
I get this error 

`Property 'entries' does not exist on type 'ObjectConstructor'.ts(2339)`

![Screen Shot 2021-04-07 at 10 43 57](https://user-images.githubusercontent.com/69221626/113895078-39743780-978e-11eb-9055-3a1cba24d491.png)

Reply:
Not sure why isn't working. What I meant to use was`Object.entries(controllers).forEach((value => value.remove());`

Another option is `Object.keys(controllers).forEach(key => controllers[key].remove());` or lodash' values